Lib/test/test_free_threading/test_enumerate.py

```diff
@@ -0,0 +1,40 @@
+import unittest
+import sys
+from threading import Thread
+
+from test.support import threading_helper
+
+
+class EnumerateThreading(unittest.TestCase):
+
+    @threading_helper.reap_threads
+    @threading_helper.requires_working_threading()
+    def test_threading(self):
+        number_of_threads = 4
+        number_of_iterations = 8
+        n = 100
+        start = sys.maxsize - 40
+        workers_enabled = False
+        def work(enum):
+            while True:
+                if workers_enabled:
+                    try:
+                        _ = next(enum)
+                    except StopIteration:
+                        break
+
+        for _ in range(number_of_iterations):
+            enum = enumerate(range(start, start + n))
+            worker_threads = []
+            for ii in range(number_of_threads):
+                worker_threads.append(
+                    Thread(target=work, args=[enum]))
+            for t in worker_threads:
+                t.start()
+            workers_enabled  = True # make sure to start all threads simultaneously
```

Comment:
Use a `threading.Barrier()` instead. For example:

https://github.com/python/cpython/blob/40a4d88a14c741172a158683c39d232c587c6f11/Lib/test/test_free_threading/test_races.py#L20-L25